arango/connections/base.py

```diff
@@ -0,0 +1,259 @@
+            host=self._host,
+            port=self._port,
+            db=self._database
+        )
+        self._username = username
+        self._password = password
+        self._http = http_client
+        self._enable_logging = enable_logging
+        self._type = 'standard'
+        self._logger = logger
+        self._async_ready = async_ready
+
+        if not async_ready:
+            self._default_job = SynchronousResultJob
+        else:
+            self._default_job = BaseJob
+
+        self._aql = AQL(self)
+        self._parent = None
+
+    def __repr__(self):
+        return '<ArangoDB connection to database "{}">'.format(self._database)
+
+    @property
+    def underlying(self):
+        if self._parent is None:
+            return self
+        else:
+            return self._parent.underlying
+
+    @property
+    def protocol(self):
+        """Return the internet transfer protocol.
+
+        :returns: the internet transfer protocol
+        :rtype: str | unicode
+        """
+        return self._protocol
+
+    @property
+    def host(self):
+        """Return the ArangoDB host.
+
+        :returns: the ArangoDB host
+        :rtype: str | unicode
+        """
+        return self._host
+
+    @property
+    def port(self):
+        """Return the ArangoDB port.
+
+        :returns: the ArangoDB port
+        :rtype: int
+        """
+        return self._port
+
+    @property
+    def username(self):
+        """Return the ArangoDB username.
+
+        :returns: the ArangoDB username
+        :rtype: str | unicode
+        """
+        return self._username
+
+    @property
+    def password(self):
+        """Return the ArangoDB user password.
+
+        :returns: the ArangoDB user password
+        :rtype: str | unicode
+        """
+        return self._password
+
+    @property
+    def database(self):
+        """Return the name of the connected database.
+
+        :returns: the name of the connected database
+        :rtype: str | unicode
+        """
+        return self._database
+
+    @property
+    def http_client(self):
+        """Return the HTTP client in use.
+
+        :returns: the HTTP client in use
+        :rtype: arango.http_clients.base.BaseHTTPClient
+        """
+        return self._http
+
+    @property
+    def logging_enabled(self):
+        """Return ``True`` if logging is enabled, ``False`` otherwise.
+
+        :returns: whether logging is enabled or not
+        :rtype: bool
+        """
+        return self._enable_logging
+
+    @property
+    def type(self):
+        """Return the connection type.
+
+        :return: the connection type
+        :rtype: str | unicode
+        """
+        return self._type
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @property
+    def async_ready(self):
+        return self._async_ready
+
+    def handle_request(self, request, handler, job_class=None,
+                       **kwargs):
```

Comment:
Please use explicit arguments instead of `**kwargs`.

Reply:
The kwargs are used exclusively for passing arguments for the job constructor. I could create an argument struct instead for this purpose, would that work?

Reply:
Which arguments are they exactly?

Reply:
response, connection, and return_result for AsyncJob, response, job_id, assign_id, and job_type for BaseJob, job_id and response for BatchJob, and job_id and response for SynchronousResultJob